File: shopping-parent/manager-web/src/main/resources/static/js/controller/specificationController.js
 //控制层 
app.controller('specificationController' ,function($scope,$controller ,specificationService){
	
	$controller('basecontroller',{$scope:$scope});//继承
	
    //读取列表数据绑定到表单中  
	$scope.findAll=function(){
		specificationService.findAll().success(
			function(response){
				$scope.list=response;
			}			
		);
	}    
	
	//分页
	$scope.findPage=function(page,rows){			
		specificationService.findPage(page,rows).success(
			function(response){
				$scope.list=response.rows;	
				$scope.paginationConf.totalItems=response.total;//更新总记录数
			}			
		);
	};

	$scope.findPageByCondition = function(){
		specificationService.findPageByCondition($scope.paginationConf.currentPage, $scope.paginationConf.itemsPerPage,$scope.entityCondition)
			.success(function (data) {
				$scope.list = data.rows;
				$scope.paginationConf.totalItems = data.total;
			})
	};
	
	//查询实体 
	$scope.findOne=function(id){				
		specificationService.findOne(id).success(
			function(response){
				$scope.entity= response;					
			}
		);				
	};
	
	//保存 
	$scope.save=function(){				
		var serviceObject = null;//服务层对象
		if($scope.entity.id!=null){//如果有ID
			serviceObject=specificationService.update; //修改
		}else{
			serviceObject=specificationService.add;//增加
		}				
		serviceObject($scope.entity)
			.success(
			function(response){
				if(response.success){
					//重新查询 
		        	$scope.reload();//重新加载
					$scope.ids = [];
					$scope.entity = {};
				}else{
					alert("失败原因："+response.message);
				}
			}		
		);				
	};

	$scope.deleteOne = function (id) {
		specificationService.deleteOne(id)
			.success(function (data) {
				if (data.success) {
					$scope.reload();
				} else {
					alert("失败原因：" + data.message);
				}
			})
	};
	 
	//批量删除
	$scope.deleteMany=function () {
		specificationService.deleteMany($scope.ids)
			.success(function (data) {
				if (data.success) {
					$scope.ids = [];
					$scope.reload();
				} else {
					$scope.ids = [];
					alert("失败原因：" + data.message);
				}
			})
	};
	
	//$scope.searchEntity={};//定义搜索对象

});	

File: shopping-parent/manager-web/src/main/java/com/nchu/manager/controller/SpecificationController.java
package com.nchu.manager.controller;
import java.util.List;

import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import com.alibaba.dubbo.config.annotation.Reference;
import com.nchu.model.Specification;
import com.nchu.sellergoods.service.SpecificationService;

import com.nchu.model.PageResult;
import com.nchu.model.Result;
/**
 * controller
 * @author Administrator
 *
 */
@RestController
@RequestMapping("/specification")
public class SpecificationController {

	@Reference
	private SpecificationService specificationService;
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findAll")
	public List<Specification> findAll(){			
		return specificationService.findAll();
	}
	
	
	/**
	 * 返回全部列表
	 * @return
	 */
	@RequestMapping("/findPage")
	public PageResult findPage(int page,int rows){
		return specificationService.findPage(page, rows,null);
	}
	
	/**
	 * 增加
	 * @param specification
	 * @return
	 */
	@RequestMapping("/add")
	public Result add(@RequestBody Specification specification){
		try {
			specificationService.add(specification);
			return new Result(true, "增加成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "增加失败");
		}
	}
	
	/**
	 * 修改
	 * @param specification
	 * @return
	 */
	@RequestMapping("/update")
	public Result update(@RequestBody Specification specification){
		try {
			specificationService.update(specification);
			return new Result(true, "修改成功");
		} catch (Exception e) {
			e.printStackTrace();
			return new Result(false, "修改失败");
		}
	}	
	
	/**
	 * 获取实体
	 * @param id
	 * @return
	 */
	@RequestMapping("/findOne")
	public Specification findOne(Long id){
		return specificationService.findOne(id);		
	}

	@RequestMapping("/deleteOne")
	public Result deleteOne(Long id){
		Result result = null;
		try {
			Integer res=specificationService.delete(id);
			if (res==0){
				result = new Result(false,"删除失败1");
			}else{
				result = new Result(true,"删除成功");
			}
		}catch (Exception e){
			result = new Result(false,"删除失败2");
		}
		return result;
	}
	
	/**
	 * 批量删除
	 * @param ids
	 * @return
	 */

	@RequestMapping("/deleteMany")
	public Result deleteMany(Long[] ids){
		Result result = null;
		try {
			Integer res=specificationService.delete(ids);
			if (res==ids.length){
				result = new Result(true,"删除成功");
			}else{
				result = new Result(false,"删除失败1");
			}
		}catch (Exception e){
			result = new Result(false,"删除失败2");
		}
		return result;
	}
	
		/**
	 * 查询+分页
	 * @param
	 * @param page
	 * @param rows
	 * @return
	 */
	@RequestMapping("/findPageByEntity")
	public PageResult<Specification> findPageByEntity(Integer page, Integer rows,@RequestBody Specification entityCondition ){
		return specificationService.findPage(page, rows,entityCondition);
	}


}
